examples/simulation/plot_simulate_sample_dataset.py

```diff
@@ -0,0 +1,161 @@
+        label.values[:] = 1
+        flip = mne.label_sign_flip(label, src)
+        lidx = np.where(np.in1d(src[hidx]['vertno'], [label.vertices]))[0]
+        lidx += offsets[hidx]
+        dotter[lidx, li] = flip / len(flip)  # normalize
+        used[lidx] = True
+    # For efficiency, restrict our actually used vertices
+    vertices = [vertices[0][used[:offsets[1]]], vertices[1][used[offsets[1]:]]]
+    n_vertices = sum(len(v) for v in vertices)
+    del offsets
+    dotter = dotter[used]
+    assert dotter.shape[0] == n_vertices
+    print('    Reduced from %d sources to %d for %d labels'
+          % (len(used), used.sum(), len(label_names)))
+
+    # Actually do the generation
+    count = 0
+    tstep = 1. / raw.info['sfreq']
+    assert events[0, 0] > raw.first_samp
+    rev_id = {val: key for key, val in event_id.items()}
+    # one signal for all for now
+    n_active = int(round(0.2 * raw.info['sfreq']))
+    signal = 1e-9 * np.hanning(n_active)
+    while True:
+        if count == 0:
+            n_samp = events[0, 0] - raw.first_samp
+            id_ = 0
+        else:
+            if count == len(events):
+                n_samp = len(raw.times) - (events[-1, 0] - raw.first_samp)
+            else:
+                assert 0 < count < len(events)
+                n_samp = events[count, 0] - events[count - 1, 0]
+            id_ = events[count - 1, 2]
+        assert n_samp > 0
+        data = np.zeros((len(label_names), n_samp))
+        if id_ not in rev_id or rev_id[id_] not in activations:
+            print('    Ignoring event %d / %d (unknown id: %d)'
+                  % (count, len(events), id_))
+        else:
+            key = rev_id[id_]
+            print('    Generating STC for event %d / %d (type %s)'
+                  % (count, len(events), key))
+            for name, amp in activations[key]:
+                row_idx = label_names.index(name)
+                time_sl = slice(0, min(len(signal), data.shape[1]))
+                data[row_idx, time_sl] = amp * signal[time_sl]
+        # Project label activations into source space
+        data = np.dot(dotter, data)
+        # Yield the SourceEstimate
+        yield (mne.SourceEstimate(data, vertices, 0, tstep), id_)
+        count += 1
+
+
+##############################################################################
+# Run the simulation
+# ------------------
+
+# Compute a noise covariance to add noise to the data
+noise_epochs = mne.Epochs(raw, events, tmax=0)
+noise_epochs.info['bads'] = []
+noise_cov = mne.compute_covariance(noise_epochs)
+raw_sim = mne.simulation.simulate_raw(
+    raw, stc_gen(activations), None, None, None, noise_cov,
+    blink=True, ecg=True, forward=forward, random_state=0, verbose=True)
+
+##############################################################################
+# Source localize
+# ---------------
+
+method, lambda2 = 'dSPM', 1. / 9.
+epochs = mne.Epochs(raw, events, event_id)
```

Comment:
raw -> raw_sim

using raw_sim you see that it does not look nice.

I still think that the current API does not do the job. If you want to use events it does not make sense to concatenate epochs as done here. I would synthesize time courses with certain durations and use make_fixed_length_events to position then along the duration of the raw file.

no rush here. We can polish this during the sprint.

my 2c

Reply:
> using raw_sim you see that it does not look nice.

What's not nice about it? It produces events at the times you asked for using a simple API. If you say that the waveforms do not look nice, that is a `SourceEstimator` issue, not a `simulate_raw` one.

> If you want to use events it does not make sense to concatenate epochs as done here. I would synthesize time courses with certain durations and use make_fixed_length_events to position then along the duration of the raw file.

I don't see it this way. You asked me how one could simulate the `sample` dataset using a simple API. This API is simple, and does what you asked. (Aside, of course, from `SourceSimulator` not being complete/specified/producing nice source time courses.) If you can see a simpler API, feel free to modify the script to at least have stubs showing how you think it could be done more simply.

But yes I can wait until the sprint to convince you in person what the (as limited as possible!) responsibilities of `simulate_raw` should be :)